tests/preflight/preflight_test.go

```diff
@@ -0,0 +1,753 @@
+			Expect(outputLogData).To(MatchRegexp("(DNS pod - dnsutils-)([a-z]{6})( hasn't reached into ready state)"))
+		})
+
+		It("Preflight check DNS resolution should fail", func() {
+			Expect(outputLogData).To(ContainSubstring("Preflight check for DNS resolution failed :: timed out waiting for the condition"))
+		})
+
+		It("Preflight check for volume snapshot and restore should fail", func() {
+			Expect(outputLogData).
+				To(MatchRegexp("(Preflight check for volume snapshot and restore failed :: pod source-pod-)" +
+					"([a-z]{6})( hasn't reached into ready state)"))
+		})
+	})
+
+	Context("When service account does not exist on cluster", func() {
+		var outputLogData string
+		var once sync.Once
+		var inputFlags = make(map[string]string)
+
+		BeforeEach(func() {
+			once.Do(func() {
+				var byteData []byte
+				copyMap(flagsMap, inputFlags)
+				inputFlags[serviceAccountFlag] = invalidServiceAccountName
+				runPreflightChecks(inputFlags)
+				byteData, err = getLogFileData(preflightLogFilePrefix)
+				Expect(err).To(BeNil())
+				outputLogData = string(byteData)
+			})
+		})
+
+		It("Preflight check for DNS resolution should fail", func() {
+			Expect(outputLogData).
+				To(MatchRegexp(fmt.Sprintf("(Preflight check for DNS resolution failed :: pods \"dnsutils-)([a-z]{6}\")"+
+					"( is forbidden: error looking up service account %s/%s: serviceaccount \"%s\" not found)",
+					defaultTestNs, invalidServiceAccountName, invalidServiceAccountName)))
+		})
+
+		It("Should not be able to create volume snapshot source pod", func() {
+			Expect(outputLogData).To(MatchRegexp(
+				fmt.Sprintf("(pods \"source-pod-)([a-z]{6})\" is forbidden: error looking up service account %s/%s: serviceaccount \"%s\" not found",
+					defaultTestNs, invalidServiceAccountName, invalidServiceAccountName)))
+		})
+
+		It("Preflight check for volume snapshot and restore should fail", func() {
+			Expect(outputLogData).To(MatchRegexp(
+				fmt.Sprintf("(Preflight check for volume snapshot and restore failed)(.*)"+
+					"(error looking up service account %s/%s: serviceaccount \"%s\" not found)",
+					defaultTestNs, invalidServiceAccountName, invalidServiceAccountName)))
+		})
+	})
+
+	Context("When incorrect log-level is provided as input", func() {
+		var outputLogData string
+		var once sync.Once
+		var inputFlags = make(map[string]string)
+
+		BeforeEach(func() {
+			once.Do(func() {
+				var byteData []byte
+				copyMap(flagsMap, inputFlags)
+				inputFlags[logLevelFlag] = invalidLogLevel
+				runPreflightChecks(inputFlags)
+				byteData, err = getLogFileData(preflightLogFilePrefix)
+				Expect(err).To(BeNil())
+				outputLogData = string(byteData)
+			})
+		})
+
+		It("Should set log-level as INFO", func() {
+			Expect(outputLogData).To(ContainSubstring("Failed to parse log-level flag. Setting log level as INFO"))
+		})
+	})
+
+	Context("Cleanup resources according to preflight UID in a particular namespace", func() {
```

Comment:
After cleanup is done, check if any resource with preflight UID label is present or not

Reply:
After performing cleanup, the resources still exist in cluster could be in terminating state.
Sometimes after executing cleanup command, the pods, PVCs still remain in running state.

@msurudwad can you please suggest an alternative approach for assertion in test, or do we need to have changes in cleanup command implementation itself.

Reply:
Used ```Eventually``` func